identity/registry/endpoints.go

```diff
@@ -67,12 +70,14 @@ type RegistrationDataDTO struct {
 type registrationEndpoint struct {
 	dataProvider   RegistrationDataProvider
 	statusProvider IdentityRegistry
+	ownIdentity    *identity.Identity
 }
 
-func newRegistrationEndpoint(dataProvider RegistrationDataProvider, statusProvider IdentityRegistry) *registrationEndpoint {
+func newRegistrationEndpoint(dataProvider RegistrationDataProvider, statusProvider IdentityRegistry, identity *identity.Identity) *registrationEndpoint {
```

Comment:
This registration data is becoming to complicated.
Maybe it's worth splitting it into two apis for service provider:
1. identities/current -> returns currently used identity address
2. identities/<id>/registration data
This way registration endpoint can be left untouched and reused

Reply:
I agree.
And soon we will have endpoint `PUT /service { providerId: "0x1" }` and caller will know his current identity